.github/workflows/publish_future.yml

```diff
@@ -0,0 +1,39 @@
+name: Deploy future tag
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  deployment:
+    name: Build and publish future tag
+    runs-on: ubuntu-latest
+    container: docker
+    permissions:
+      id-token: write
+      contents: write
+      packages: write
+
+    steps:
+      - name: Get latest tag
```

Comment:
I don't think you want this. The future tag should be built from `main`, right?

Reply:
You're mistaking this for `checkout_ref` variable

Reply:
Ah ok - but in that case surely it's not required? It looks like it's only used when creating the GitHub release, and you're not creating a release when the title is empty.

Maybe just create an input called `publish_release`, set it to `false` for the future build and true for the others, and don't make `git_tag` required? That way it's obvious what's happening.

Reply:
Makes sense! Refactored it! Thanks